cbmc/proofs/mqtt_state.c

```diff
@@ -228,7 +228,10 @@ bool valid_IotMqttConnection( const IotMqttConnection_t pConn )
   // maximum value to avoid integer overflows.  We expect to run out of
   // memory before having 2^16 references on a device.
   bool valid_references =
-    pConn->references >= 1 &&
+    ( (pConn->pingreq.u.operation.periodic.ping.keepAliveMs == 0 &&
+       pConn->references > 0) ||
+      (pConn->pingreq.u.operation.periodic.ping.keepAliveMs != 0 &&
+       pConn->references > 1) ) &&
```

Comment:
Is this true in general?  Can you add a comment pointing to the documentation validating this assumption?  And wouldn't it be better to formulate these as implication (using the IMPLIES macro): (keepalive == 0 => references > 0) && (keepalive != 0 => references > 1)

Reply:
This is in general and it was reported directly by @gordonwang0. Using IMPLIES is cleaner!